src/lib.rs

```diff
@@ -302,34 +207,55 @@ pub mod pipeline {
             // after we match (staying as Vec<u8> until then)
             let mut line = String::with_capacity(1024);
             while buf.read_line(&mut line).unwrap() > 0 {
-                if let Some(row) = Pipeline::proc_preagg(&line, &self.filter, &preaggs) {
-                    tx.send(row).unwrap();
+                if self.filter.iter().all(|re| re.is_match(&line)) {
+                    if !Pipeline::proc_preagg(Record::new(&line), &mut preaggs, &tx) {
+                        // One of the operators has closed, break out of the loop.
```

Comment:
If a limit is after an aggregate, it should be in an adapter where it will be re-instantiated and only the output of the aggregate will be limited.  So, in your count example, only two lines are displayed, which is "right" (I think).  But, the output of the count is wrong, not sure why yet.